internal/reminder/reminder.go

```diff
@@ -149,70 +137,77 @@ func (r *reminder) Stop() {
 	})
 }
 
-// storeCursorState stores the cursor state to a file
-// Not thread-safe, should be called from a single goroutine
-func (r *reminder) storeCursorState(ctx context.Context) error {
+func (r *reminder) sendReminders(ctx context.Context) []error {
 	logger := zerolog.Ctx(ctx)
-	logger.Debug().Msg("storing cursor state")
-
-	var buf bytes.Buffer
-	enc := gob.NewEncoder(&buf)
 
-	data := map[string]interface{}{
-		"projectListCursor": r.projectListCursor,
-		"repoListCursor":    r.repoListCursor,
+	// Fetch a batch of repositories
+	repos, err := r.getRepositoryBatch(ctx)
+	if err != nil {
+		logger.Error().Err(err).Msg("unable to fetch repositories")
+		return []error{err}
 	}
-	if err := enc.Encode(data); err != nil {
-		return err
+
+	logger.Info().Msgf("created repository batch of size: %d", len(repos))
+
+	// Update the reminder_last_sent for each repository to export as metrics
+	for _, repo := range repos {
+		logger.Debug().Msgf("updating reminder_last_sent for repository: %s", repo.ID)
```

Comment:
Zerolog supports structured logging, which allows us to later easily query and extract (for example) log lines that reference a specific repository id:

```suggestion
		logger.Debug().Str("repo", repo.ID.String()).Time("previously", repo.ReminderLastSent.Time).
			Msg("updating reminder_last_sent")
```

Reply:
Done